pkg/api/controller.go

```diff
@@ -4830,3 +4838,18 @@ func encodeGCUncommittedMark(mark *catalog.GCUncommittedMark) (*string, error) {
 	token := base64.StdEncoding.EncodeToString(raw)
 	return &token, nil
 }
+
+func extractLakeFSMetadata(header http.Header) map[string]string {
+	meta := make(map[string]string)
+	for k, v := range header {
+		lowerKey := strings.ToLower(k)
+		if strings.HasPrefix(lowerKey, LakeFSHeaderMetadataPrefix) {
+			_, metaKey, _ := strings.Cut(strings.ToLower(k), LakeFSHeaderMetadataPrefix)
+			meta[metaKey] = v[0]
+		} else if strings.HasPrefix(lowerKey, LakeFSHeaderInternalPrefix) {
+			_, metaKey, _ := strings.Cut(strings.ToLower(k), LakeFSHeaderInternalPrefix)
+			meta[LakeFSMetadataPrefix+metaKey] = v[0]
+		}
+	}
+	return meta
```

Comment:
- reuse lowerKey instead of calling to lower again
- substring as we already matched the prefix

```suggestion
	meta := make(map[string]string)
	for k, v := range header {
		lowerKey := strings.ToLower(k)
		metaKey := ""
		if strings.HasPrefix(lowerKey, LakeFSHeaderMetadataPrefix) {
			metaKey = lowerKey[len(LakeFSHeaderMetadataPrefix):]
		} else if strings.HasPrefix(lowerKey, LakeFSHeaderInternalPrefix) {
			metaKey = LakeFSMetadataPrefix + lowerKey[len(LakeFSHeaderInternalPrefix):]
		} else {
			continue
		}
		meta[metaKey] = v[0]
	}
	return meta
```